File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string textMain;
    textMain = get_string("Enter text: ");

    int characterCounter = 0;
    int letterCounter = 0;
    int wordCounter = 1;
    int sentenceCounter = 0;
    
    characterCounter = strlen(textMain);
    
    int i = 0;
    for (i = 0; characterCounter > i; i++)
    {
        if ((textMain[i] >= 'a' && textMain[i] <= 'z') || (textMain[i] >= 'A' && textMain[i] <= 'Z'))
        {
            letterCounter++;
        }
        if (textMain[i] == ' ')
        {
            wordCounter++;
        }
        if (textMain[i] == '.' || textMain[i] == '!' || textMain[i] == '?')
        {
            sentenceCounter++;
        }
    }
    //printf("%i character(s)\n", characterCounter);
    //printf("%i letter(s)\n", letterCounter);
    //printf("%i word(s)\n", wordCounter);
    //printf("%i sentence(s)\n", sentenceCounter);

    float L = 0;
    float S = 0;
    
    L = ((float)letterCounter / (float)wordCounter) * 100;
    S = ((float)sentenceCounter / (float)wordCounter) * 100;
    //printf("%f, %f\n", L, S);
    
    int index = 0;
    index = round((0.0588 * L) - (0.296 * S) - 15.8);
    
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 1 && index <= 16)
    {
        printf("Grade %i\n", index);
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
     

}

File: pset6/readability/readability.py
from cs50 import get_string

# put text to check in var
textMain = get_string("Enter text: ")

# var needed for this ex
characterCounter = 0
letterCounter = 0
wordCounter = 1
sentenceCounter = 0

# number of char
characterCounter = len(textMain)

# number of letter
for isLetter in textMain:
    if isLetter.isalpha():
        letterCounter += 1
        
# number of word
for isSpace in textMain:
    if (isSpace == " "):
        wordCounter += 1
        
# number of sentence
for isSentence in textMain:
    if (isSentence == "." or isSentence == "!" or isSentence == "?"):
        sentenceCounter += 1
      
# print(characterCounter)
# print(letterCounter)
# print(wordCounter)
# print(sentenceCounter)

L = letterCounter * 100 / wordCounter
S = sentenceCounter * 100 / wordCounter

# Coleman-Liau index
indexColemanLiau = round(0.0588 * L - 0.296 * S - 15.8)

# check what grade
if (indexColemanLiau < 1):
    print("Before Grade 1")

elif (indexColemanLiau >= 16):
    print("Grade 16+")

else:
    print("Grade ", indexColemanLiau)